### PR TITLE
fix(windows): improve support for strings.xml

### LINKED ISSUE #4264: bug(windows): backslash before the double quotes
**Describe the bug**

A backslash is found before each double quote in the strings. I presume it is not what is expected.

Windows-10_Keyman-14 0 222-beta_backslash

**Expected behavior**

No backslash should be placed before any double quotes.

---

**Keyman for Windows:**
 - OS: Windows 10 Pro 20H2 x64
 - Keyman version: 14.0.222-beta
 - Target Application: self

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #4264.

The Android strings.xml format that we now use with Keyman Desktop has a number of escapes that must be supported in order for text strings to be rendered correctly, including: @, ?, ', ". Furthermore, positional parameters have a slightly different format which we should transform on load.

This change includes a unit test androidstringtokeymanlocalestring.

### CODE REVIEW

File: windows/src/global/delphi/general/Keyman.System.AndroidStringToKeymanLocaleString.pas [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    end
+    else if (i <= text.Length - 3) and
+      (text[i] = '%') and
+      (text[i+1] >= '1') and (text[i+1] <= '9') and
+      (text[i+2] = '$') and
+      CharInSet(text[i+3], ['d','u','e','f','g','n','m','p','s','x']) then
+    begin
+      // 4. Transform positional parameters
+      // convert from %1$d to %0:d; Delphi parameters are 0-based
+      // note: only supports up to 9 positional parameters
```
Comment (darcywong00) [inline windows/src/global/delphi/general/Keyman.System.AndroidStringToKeymanLocaleString.pas:53]: Is the limit of 9 positional parameters made for practicality purposes?

Can maybe add a unit test for this boundary so we'll see a fail if someone tries to make a string with 10 params

File: windows/src/test/unit-tests/androidstringtokeymanlocalestring/Keyman.Test.AndroidStringToKeymanLocaleStringTest.pas [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+procedure TAndroidStringToKeymanLocaleStringTest.TestTransform;
+begin
+  Assert.AreEqual('', TAndroidStringToKeymanLocaleString.Transform(''));
+  Assert.AreEqual('text', TAndroidStringToKeymanLocaleString.Transform('text'));
+  Assert.AreEqual('text', TAndroidStringToKeymanLocaleString.Transform('"text"'));
+  Assert.AreEqual('te"xt', TAndroidStringToKeymanLocaleString.Transform('te\"xt'));
+  Assert.AreEqual('te'#13#10'xt', TAndroidStringToKeymanLocaleString.Transform('te\nxt'));
+  Assert.AreEqual('1\2?3@4', TAndroidStringToKeymanLocaleString.Transform('1\\2\?3\@4'));
+  Assert.AreEqual('%2:s text %0:d %1:s', TAndroidStringToKeymanLocaleString.Transform('%3$s text %1$d %2$s'));
```
Comment (darcywong00) [inline windows/src/test/unit-tests/androidstringtokeymanlocalestring/Keyman.Test.AndroidStringToKeymanLocaleStringTest.pas:29]: What about testing other formats?
~'d'~, 'u', 'e', 'f', 'g', 'n', 'm', 'p', ~'s'~, 'x'

**Comment by darcywong00** [comment]
The changes lgtm. I'll approve when the failing Desktop build passes

File: windows/src/global/delphi/general/Keyman.System.AndroidStringToKeymanLocaleString.pas (hunk 1 above)
Comment (mcdurdin) [reply windows/src/global/delphi/general/Keyman.System.AndroidStringToKeymanLocaleString.pas:53]: I don't think it's necessary. We use at most 2 parameters currently. It'd be overkill to create a test against all the strings.xml files anyway to exercise that!

File: windows/src/test/unit-tests/androidstringtokeymanlocalestring/Keyman.Test.AndroidStringToKeymanLocaleStringTest.pas (hunk 2 above)
Comment (mcdurdin) [reply windows/src/test/unit-tests/androidstringtokeymanlocalestring/Keyman.Test.AndroidStringToKeymanLocaleStringTest.pas:29]: I don't think we have to check every flavour -- it doesn't improve the quality of the test case, really.

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 14.0.229-beta](https://keyman.com/downloads/releases/beta/14.0.229)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 14.0.230-beta](https://keyman.com/downloads/releases/beta/14.0.230)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.15-alpha](https://keyman.com/downloads/releases/alpha/15.0.15)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.16-alpha](https://keyman.com/downloads/releases/alpha/15.0.16)
